src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -449,25 +462,29 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         _tenantList.clear();
         for (const auto& entry : credList)
         {
-            auto nameJson = json::value::parse(entry.UserName().c_str());
-            std::optional<int> credentialVersion;
-            if (nameJson.has_integer_field(U("ver")))
+            try
             {
-                credentialVersion = nameJson.at(U("ver")).as_integer();
-            }
+                auto nameJson = WDJ::JsonObject::Parse(entry.UserName());
+                std::optional<int> credentialVersion;
+                if (nameJson.HasKey(L"ver"))
+                {
+                    credentialVersion = static_cast<int>(nameJson.GetNamedNumber(L"ver"));
+                }
 
-            if (!credentialVersion.has_value() || credentialVersion.value() != CurrentCredentialVersion)
-            {
-                // ignore credentials that aren't from the latest credential revision
-                vault.Remove(entry);
-                oldVersionEncountered = true;
-                continue;
-            }
+                if (!credentialVersion.has_value() || credentialVersion.value() != CurrentCredentialVersion)
+                {
+                    // ignore credentials that aren't from the latest credential revision

```

Comment:
I could just solve the duplication TODO by bumping the version. How do y'all feel about that?

Reply:
Chose to do it. This invalidates all old credentials.